js/src/hooks/useStoreAddress.js

```diff
@@ -52,18 +52,18 @@ export default function useStoreAddress() {
 
 	if ( loaded && contact ) {
 		const {
-			wc_address: {
-				country: countryCode,
-				locality: city,
-				postal_code: postcode,
-				region: state,
-				street_address: streetAddress,
-			},
+			wc_address: wcAddress,
 			is_mc_address_different: isMCAddressDifferent,
 		} = contact;
 
+		// Handle fallback for `null` fields to make sure the returned data types are consistent.
+		const streetAddress = wcAddress.street_address || '';
+		const city = wcAddress.locality || '';
+		const state = wcAddress.region || '';
+		const postcode = wcAddress.postal_code || '';
+
 		const [ address, address2 = '' ] = streetAddress.split( '\n' );
-		const country = countryNameDict[ countryCode ];
+		const country = countryNameDict[ wcAddress.country ];
```

Comment:
I thought about whether want to specify fallback `''` here in case `wcAddress.country` does not exist in `countryNameDict`, but in the end that is not necessary because users can only select a value from the predefined values in the country dropdown and cannot enter a value of their own, so there shouldn't be any problem.